runtime/client/kubeconfig.go

```diff
@@ -0,0 +1,94 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"time"
+
+	"k8s.io/client-go/rest"
+)
+
+// KubeConfigOptions defines options for KubeConfig sanitization.
+type KubeConfigOptions struct {
```

Comment:
We need to make the Kubernetes API address configurable. For example, Kind creates kubeconfigs using localhost, for Flux to connect from one Kind cluster to another, the server address must be changed before creating the secret. Once we have the address as an option, then we can implement `flux create secret kubeconfig --address https://172.20.0.3:6443/ -f ./my-cluster.yaml`.

Reply:
I gave some thought to this and I think we should leave the address out for now. The CLI will also need the service account name, but all these options are not needed for the controllers.

Reply:
Added the BindFlags here to accommodate the key options as discussed. PTAL